src/components/atoms/FormModal/FormModal.tsx

```diff
@@ -37,17 +39,34 @@ const FormModal = ({
   disableSubmit,
   loadingSubmit,
   loadingContent,
+  onFinish,
   form,
   ...formProps
 }: FormModalProps) => {
   const { t } = useTranslation();
 
+  const { setIsSubmitting } = useUnsavedPrompt({ form });
+
   useEffect(() => {
     if (!visible) {
       form?.resetFields();
     }
   }, [form, visible]);
 
+  useEffect(() => {
+    if (!visible) {
+      setIsSubmitting(false);
+    }
+  }, [setIsSubmitting, visible]);
```

Comment:
these two useEffects could be combined right? 🤔

Reply:
I think the first one is no longer needed. Because I now reset the fields on submit and also after confirming the unsaved prompt. I will remove it.